zookeeper-server/src/main/java/org/apache/zookeeper/server/auth/X509AuthenticationConfig.java

```diff
@@ -300,6 +308,17 @@ public String getZnodeGroupAclClientUriDomainMappingRootPath() {
     return ZNODE_GROUP_ACL_CLIENTURI_DOMAIN_MAPPING_ROOT_PATH;
   }
 
+  private Set<String> loadSuperUserIds() {
+    if (znodeGroupAclSuperUserId == null) {
+      setZnodeGroupAclSuperUserId(System.getProperty(ZOOKEEPER_ZNODEGROUPACL_SUPERUSER_ID));
+    }
+    if (znodeGroupAclSuperUserId == null || znodeGroupAclSuperUserId.isEmpty()) {
```

Comment:
If we keep only one then all these things would go away

Reply:
The way it is implemented is because we are doing lazy instantiation for both the znodeGroupAclSuperUserIdStr raw string and znodeGroupAclSuperUserIds set, the reason is that this config has two sources: Zookeeper config and system properties.